src/Elastic.Apm/Metrics/MetricsProvider/ProcessWorkingSetAndVirtualMemoryProvider.cs

```diff
@@ -0,0 +1,33 @@
+using System.Collections.Generic;
+using System.Diagnostics;
+using Elastic.Apm.Api;
+
+namespace Elastic.Apm.Metrics.MetricsProvider
+{
+	public class ProcessWorkingSetAndVirtualMemoryProvider : IMetricsProvider
+	{
+		private const string ProcessVirtualMemory = "system.process.memory.size";
+		private const string ProcessWorkingSetMemory = "system.process.memory.rss.bytes";
+
+
+		public int ConsecutiveNumberOfFailedReads { get; set; }
+		public string NameInLogs => "process's working set and virtual memory size";
+
+		public IEnumerable<Sample> GetValue()
+		{
+			var process = Process.GetCurrentProcess();
+			var virtualMemory = process.VirtualMemorySize64;
+			var workingSet = process.WorkingSet64;
+
+			var retVal = new List<Sample>();
+
+			if (virtualMemory != 0)
```

Comment:
Just looks like wrong value.

Reply:
But if it actually happens how will we know why agent doesn't report this metric at all?

Reply:
Happy to change in case you have a suggestion. 

By looking at the rest of the code I think it's clear that it can only be caused by `Process.VirtualMemorySize64`/`Process.WorkingSet64` returning 0, which we can't do much about, since those are part of CoreFX. 
What else should we do? 

Sending 0 sounds bad, since those are with high probability invalid values, adding a log would add more noise (we log the fact that this did not work anyway in `MetricsCollector` if it happens 5 times subsequently), I don't see any better way.

Reply:
If the only reason `Process.VirtualMemorySize64/Process.WorkingSet64` can return  0 is a bug in CoreFX then I don't think we try to hide it. I propose to use whatever `Process.VirtualMemorySize64/Process.WorkingSet64` return directly and deal with a bug in CoreFX when we get there.

Reply:
So you want to send a metrics with 0kb available memory? 

Also I'd be careful with saying 
> If the only reason Process.VirtualMemorySize64/Process.WorkingSet64 can return 0 is a bug in CoreFX

It can be anything, those CoreFx methods usually wrap OS calls (like Win32 on windows) - those can also return 0, so not necessarily a CoreFx bug.. maybe on some strange OS it's defined to return 0 in some scenarios.

Reply:
Googling for "problems with Process.VirtualMemorySize64" ... 
One of the hits - https://github.com/dotnet/corefx/issues/20812
Checking for zero would not have helped...
It's like trying to deal with CPU bugs after https://en.wikipedia.org/wiki/Pentium_FDIV_bug - I don't see any point in spending the effort.

Reply:
Or https://github.com/dotnet/corefx/issues/20985 - again checking for 0 would not have helped.

Reply:
Did not think that a check for `0` will solve all potential issues 😅 

So what do you want me to do? Remove the check for `0`? 

_(also for the record: again, I think having it there is fine, but just for reaching an agreement I accept everything - I seriously think we are not productive here...)_

Reply:
Yes, I think there is not point in checking for zero.
_I would be glad to discuss if you feel we can improve our review process._

Reply:
_I think we definetely should after this :)_ 

Ok, removing it 😞 .